src/test/scala/io/iohk/ethereum/mpt/MerklePatriciaTrieSuite.scala

```diff
@@ -555,42 +555,25 @@ class MerklePatriciaTrieSuite extends AnyFunSuite with ScalaCheckPropertyChecks
 
   test("getProof returns valid proof for existing key") {
     import scala.util.Random
+    import MptProofVerifier.verifyProof
 
-    forAll(Gen.nonEmptyListOf(Arbitrary.arbitrary[(Int, Int)])) { keyValueList: Seq[(Int, Int)] =>
-      // given
+    forAll(keyValueListGen()) { keyValueList: Seq[(Int, Int)] =>
       val input: Seq[(Array[Byte], Array[Byte])] = keyValueList
         .map { case (k, v) => k.toString.getBytes() -> v.toString.getBytes() }
 
-      val keyToFind: Array[Byte] = input.headOption
-        .getOrElse(fail("Cant check proof for empty collection"))
-        ._1
-
-      val trie = Random
-        .shuffle(input)
+      val trie = Random.shuffle(input)
```

Comment:
shuffle is not needed here, right? we are testing all the values from 'input' anyway

Reply:
True. Removed.